Indoor_GPS/marvelmind.py

```diff
@@ -108,14 +110,24 @@ def print_position(self):
             print ("Hedge {:d}: X: {:.3f}, Y: {:.3f}, Z: {:.3f}, Angle: {:d} at time T: {:.2f}".format(self.position()[0], self.position()[1], self.position()[2], self.position()[3], self.position()[4], self.position()[5]/1000.0))
 
     def position(self):
-        return list(self.valuesUltrasoundPosition)[-1];
+        ans = list(self.valuesUltrasoundPosition)[-1]
+        if self.num_called < 5:
+            if self.num_called == 0:
+                self.zero_pos = np.array(ans)
+            else:
+                self.zero_pos = ((self.num_called) * self.zero_pos + np.array(ans))/(self.num_called + 1)
```

Comment:
Cute math, i like

Reply:
What exactly is this if/else statement doing?